services/f-droid/f-droid.service.js

```diff
@@ -0,0 +1,74 @@
+'use strict'
+
+const BaseHTTPService = require('../base-http')
+const { addv: versionText } = require('../../lib/text-formatters')
+const { version: versionColor } = require('../../lib/color-formatters')
+const { InvalidResponse } = require('../errors')
+
+module.exports = class FDroid extends BaseHTTPService {
+  async fetch({ appId }) {
+    const url = `https://f-droid.org/en/packages/${appId}/`
+    return this._requestHTTP({
+      url,
+      options: {},
+      errorMessages: {
+        404: 'app not found',
+      },
+    }).then(({ res, buffer }) => {
+      const website = buffer.toString()
+      // we assume the layout as provided here:
```

Comment:
As mentioned in #1963, I don't think not having an API is great. I'm personally fine with this approach in the meantime, there will probably not be a high volume of F-Droid requests hitting our servers initially, especially with the static example and the extra caching we've now got in place. Nevertheless I think we should open a proper API feature request on the F-Droid and add the link here.

Reply:
Sounds good. Who do you think could do that best?

Reply:
I added a comment here: https://gitlab.com/fdroid/fdroiddata/issues/1325#note_96914686

Reply:
We've generally declined badges that require scraping. There are many examples of this is one exception, and one exception which was added several years ago.

Is it possible to push forward on building the fdroid API?

I am open to scraping in principle; however we should do a few things:

1. Build in more server capacity
2. Use cheerio or something similar; because [parsing html using regex is problematic](https://stackoverflow.com/questions/1732348/regex-match-open-tags-except-xhtml-self-contained-tags/1732454#1732454).
3. Cache them appropriately
4. Isolate the performance of these badges in some way so that, if they get slow, they only affect the other scraping badges

Reply:
It appears we could pull the data we need from the "build metadata" file that's linked from the package web page.

e.g. https://gitlab.com/fdroid/fdroiddata/blob/master/metadata/org.pacien.tincapp.txt

@niccokunzmann Could you please update this to read from that file?